### PR TITLE
feat(menu): add header and action items slots (VIV-947)

### PR DESCRIPTION
**Pull request by rinaok**

### CODE REVIEW

File: libs/components/src/lib/menu/README.md [hunk 1]
```diff
@@ -106,6 +116,65 @@ While any DOM content is permissible as a child of the menu, only `vwc-menu-item
...
+</style>
+
+<vwc-menu open>
+  <vwc-text-field slot="header" placeholder="Search" icon="search"></vwc-text-field>
+  <vwc-menu-item text="Menu item 1"></vwc-menu-item>
+  <vwc-menu-item text="Menu item 2"></vwc-menu-item>
+</vwc-menu>
+```
+
+### Action Items
```
Comment (rachelbt) [inline libs/components/src/lib/menu/README.md:137]: why do we nned both action-items and footer?

File: libs/components/src/lib/menu/menu.scss [hunk 2]
```diff
@@ -2,6 +2,12 @@
 @use "partials/variables" as variables;
 
 .base {
+	display: flex;
+	box-sizing: border-box;
+	flex-direction: column;
```
Comment (rachelbt) [inline libs/components/src/lib/menu/menu.scss:7]:
```suggestion
	flex-direction: column;
	gap: 8px;
```

File: libs/components/src/lib/menu/menu.scss [hunk 3]
```diff
@@ -22,3 +28,29 @@
 	outline: 2px solid currentColor;
 	outline-offset: -2px;
 }
+
+.header {
+	padding: 24px 24px 0;
```
Comment (rachelbt) [inline libs/components/src/lib/menu/menu.scss:33]:
```suggestion
	padding-inline: 16px;
	padding-block-start: 8px;
```

File: libs/components/src/lib/menu/menu.scss [hunk 4]
```diff
@@ -2,6 +2,12 @@
...
 .base {
+	display: flex;
+	box-sizing: border-box;
+	flex-direction: column;
+}
+
+.body {
 	max-block-size: var(#{variables.$menu-block-size}, 408px);
 	max-inline-size: var(#{variables.$menu-max-inline-size}, 100%);
 	min-inline-size: var(#{variables.$menu-min-inline-size});
```
Comment (rachelbt) [inline libs/components/src/lib/menu/menu.scss:13]: I'm not sure this need to be here - this needs to be on the base

Comment (rachelbt) [reply libs/components/src/lib/menu/menu.scss:13]: also the overflow

Comment (rachelbt) [reply libs/components/src/lib/menu/menu.scss:13]: and remove the `padding-block: 8px;`

File: libs/components/src/lib/menu/menu.scss [hunk 5]
```diff
@@ -22,3 +28,29 @@
...
+	.hide-header & {
+		display: none;
+	}
+}
+
+.footer {
+	display: flex;
+	align-items: center;
+	justify-content: space-between;
+	padding: 0 24px 24px;
```
Comment (rachelbt) [inline libs/components/src/lib/menu/menu.scss:44]:
```suggestion
	padding-inline: 16px;
	padding-block-end: 8px;
```

File: libs/components/src/lib/menu/menu.template.ts [hunk 6]
```diff
@@ -37,14 +54,27 @@ export const MenuTemplate: (
...
 				</div>
+				<div class="footer">
+					<div>
+						<slot name="footer" ${slotted('footerSlottedContent')}></slot>
+					</div>
+					<div class="actions">
+						<slot name="action-items" ${slotted('actionItemsSlottedContent')}></slot>
+					</div>
+				</div>
+			</div>
```
Comment (rachelbt) [inline libs/components/src/lib/menu/menu.template.ts:77]: I don't think this is needed - we only need a footer slot.
maybe a validation components for cancel submit is needed...\

File: libs/components/src/lib/menu/README.md [hunk 7]
```diff
@@ -106,6 +116,65 @@ While any DOM content is permissible as a child of the menu, only `vwc-menu-item
...
+<style>
+  html { /* for demo purposes */
+		block-size: 200px;
+	}
+</style>
+
+<vwc-menu open>
+  <vwc-menu-item text="Menu item 1"></vwc-menu-item>
+  <vwc-menu-item text="Menu item 2"></vwc-menu-item>
+  <vwc-checkbox slot="footer" label="I agree"></vwc-checkbox>
```
Comment (rachelbt) [inline libs/components/src/lib/menu/README.md:172]:
```suggestion
```
Not sure this has to do with the menu-items :)

**Review by rachelbt — COMMENTED** [review]
went over the template and the css - left some comments :)

File: libs/components/src/lib/menu/README.md (hunk 1 above)
Comment (rinaok) [reply libs/components/src/lib/menu/README.md:137]: Similar to dialog. The examples all had action items at the bottom. Do you think we don't need it?

File: libs/components/src/lib/menu/menu.template.ts (hunk 6 above)
Comment (rinaok) [reply libs/components/src/lib/menu/menu.template.ts:77]: @rachelbt But the same as in dialog all the examples have two action-buttons at the bottom :( 
https://docs.google.com/document/d/1Ag5Nmu53-v6YgqWpN_Uz_K4-KSOYWzRGLhWqrvT6GXU/edit#heading=h.36st4rl33pcx

File: libs/components/src/lib/menu/README.md (hunk 7 above)
Comment (rinaok) [reply libs/components/src/lib/menu/README.md:172]: @rachelbt Just wanted to show an example for the footer combined with the action buttons, do you have another idea?

File: libs/components/src/lib/menu/menu.scss [hunk 8]
```diff
@@ -2,12 +2,15 @@
...
+	display: flex;
+	box-sizing: border-box;
+	flex-direction: column;
+	gap: 8px;
 	max-block-size: var(#{variables.$menu-block-size}, 408px);
 	max-inline-size: var(#{variables.$menu-max-inline-size}, 100%);
 	min-inline-size: var(#{variables.$menu-min-inline-size});
 	overflow-x: hidden;
 	overflow-y: auto;
-	padding-block: 8px;
```
Comment (RichardHelm) [inline libs/components/src/lib/menu/menu.scss:10]: The menu without header/footer is now missing the padding: 
image

**Review by RichardHelm — CHANGES_REQUESTED** [review]

File: libs/components/src/lib/menu/README.md [hunk 9]
```diff
@@ -188,3 +257,27 @@ Menu component is a low level element, unaware of its document context, but is,
 A common practice used in apps / frameworks to promote a Menu component to top other elements z-axis, is to utilise a service that dynamically appends a Menu component to the end of the body element, when called for.
 
 This helps ensure elements don't render over top a Menu undesirebly.
+
+## Use Cases
```
Comment (RichardHelm) [inline libs/components/src/lib/menu/README.md:261]: Can I put arbitrary content in default slot? Like data-grid?
image

Can we add a use case for that please?

Comment (rinaok) [reply libs/components/src/lib/menu/README.md:261]: @RichardHelm This is an example from vivid-2. We don't have checkboxes inside a data-grid in vivid-3. And also, I'm not sure if it is correct to do it. We have menu-items for checkboxes.

File: libs/components/src/lib/menu/README.md (hunk 1 above)
Comment (rinaok) [reply libs/components/src/lib/menu/README.md:137]: Updated to only action-items, removed the footer

File: libs/components/src/lib/menu/README.md (hunk 9 above)
Comment (RichardHelm) [reply libs/components/src/lib/menu/README.md:238]: It's just an example. What I mean is you are supposed to be able to put anything inside menu, right?
And if you do that, will it break accessibility?

So we should have an example for that.

E.g. in dashboard we have this (nav-items):
image

**Review by rachelbt — COMMENTED** [review]
css + HTML looks good!
The only thing that worth considering in a different ticket is the scroll - if there's a need for sticky header or footer or both.

**Review by RichardHelm — APPROVED** [review]

**Pull request merged**